pallets/dapp-staking/src/lib.rs

```diff
@@ -1521,12 +1521,12 @@ pub mod pallet {
 
             Self::update_ledger(&account, ledger)?;
 
-            // Return the `StakeAmount` that has max total value - that's the one that is equal to the `amount` parameter.
+            // Return the `StakeAmount` that has max total value
             let unstake_amount = stake_amount_iter
                 .iter()
                 .max_by(|a, b| a.total().cmp(&b.total()))
-                .expect("At least one value exists, otherwise we wouldn't be here.");
-            assert_eq!(unstake_amount.total(), amount);
```

Comment:
maybe better to add also this check to ensure values are equal?

Reply:
This is checked in two places in the test-utils used for [unstake](https://github.com/AstarNetwork/Astar/blob/0c7a75e49f1e79d711d707d312c4e94a3a579053/pallets/dapp-staking/src/test/testing_utils.rs#L1800) and [move](https://github.com/AstarNetwork/Astar/blob/0c7a75e49f1e79d711d707d312c4e94a3a579053/pallets/dapp-staking/src/test/testing_utils.rs#L859) operation tests. We might be missing a scenario, but I wanted to remove assert checks from the runtime.